base/datafmt.jl

```diff
@@ -13,28 +13,15 @@ export countlines, readdlm, readcsv, writedlm, writecsv
 const invalid_dlm = Char(0xfffffffe)
 const offs_chunk_size = 5000
 
-countlines(nameorfile) = countlines(nameorfile, '\n')
-function countlines(filename::AbstractString, eol::Char)
-    open(filename) do io
-        countlines(io, eol)
-    end
-end
-function countlines(io::IO, eol::Char)
-    if !isascii(eol)
-        throw(ArgumentError("only ASCII line terminators are supported"))
-    end
+countlines(f::AbstractString,eol::Char='\n') = open(io->countlines(io,eol),f)::Int
```

Comment:
Interesting tidbit for others out there: while the `open(func, file)` can be quick and tidy, it's also quite subtle from a type stability perspective. Without the `::Int` assertion here, `@code_warntype` shows that the return type for this method is `Any`, even though `countlines` itself is type-stable.